packages/runtime/core/src/address.ts

```diff
@@ -2,19 +2,27 @@ import * as t from "io-ts/lib/index.js";
 import { iso, Newtype } from "newtype-ts";
 import { fromNewtype } from "io-ts-types";
 import { TxOutRef } from "./tx/outRef.js";
+import { unsafeEither } from "@marlowe.io/adapter/fp-ts";
 
-export type AddressBech32 = Newtype<
-  { readonly AddressBech32: unique symbol },
-  string
->;
-export const AddressBech32 = fromNewtype<AddressBech32>(t.string);
-export const unAddressBech32 = iso<AddressBech32>().unwrap;
-export const addressBech32 = iso<AddressBech32>().wrap;
+export interface AddressBech32Brand {
+  readonly AddressBech32: unique symbol;
+}
+
+export const AddressBech32Guard = t.brand(
+  t.string,
+  (s): s is t.Branded<string, AddressBech32Brand> => true,
+  "AddressBech32"
+);
+
+export type AddressBech32 = t.TypeOf<typeof AddressBech32Guard>;
+
+export const addressBech32 = (s: string) =>
+  unsafeEither(AddressBech32Guard.decode(s));
```

Comment:
this is not needed (and thus the dependency to adapter)
This screenshot is from the old code (that still uses t.brand) and it works fine

![Screenshot 2023-12-13 at 11 01 56](https://github.com/input-output-hk/marlowe-ts-sdk/assets/2634059/26e01180-ced0-4a80-9c45-6fbef8cd463e)

Reply:
in there, x is a Branded type